From patchwork Fri Nov  9 13:59:49 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Vitaly Kuznetsov <vkuznets@redhat.com>
X-Patchwork-Id: 10675973
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id EA0A814BD
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Nov 2018 13:59:53 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id D77BB2E66C
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Nov 2018 13:59:53 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id CA2AA2EBB9; Fri,  9 Nov 2018 13:59:53 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.2 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI,SUBJ_OBFU_PUNCT_FEW autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7ABD12E66C
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri,  9 Nov 2018 13:59:53 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727845AbeKIXkf (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 9 Nov 2018 18:40:35 -0500
Received: from mx1.redhat.com ([209.132.183.28]:31132 "EHLO mx1.redhat.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1727735AbeKIXkf (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 9 Nov 2018 18:40:35 -0500
Received: from smtp.corp.redhat.com (int-mx04.intmail.prod.int.phx2.redhat.com
 [10.5.11.14])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mx1.redhat.com (Postfix) with ESMTPS id CCC3D81F09;
        Fri,  9 Nov 2018 13:59:51 +0000 (UTC)
Received: from vitty.brq.redhat.com (unknown [10.43.2.9])
        by smtp.corp.redhat.com (Postfix) with ESMTP id C4C555D968;
        Fri,  9 Nov 2018 13:59:50 +0000 (UTC)
From: Vitaly Kuznetsov <vkuznets@redhat.com>
To: kvm@vger.kernel.org
Cc: Paolo Bonzini <pbonzini@redhat.com>,
        Roman Kagan <rkagan@virtuozzo.com>
Subject: [PATCH kvm-unit-tests] x86/hyperv: pass hv_vpindex cpu flag along
 with hv_synic
Date: Fri,  9 Nov 2018 14:59:49 +0100
Message-Id: <20181109135949.24182-1-vkuznets@redhat.com>
X-Scanned-By: MIMEDefang 2.79 on 10.5.11.14
X-Greylist: Sender IP whitelisted,
 not delayed by milter-greylist-4.5.16 (mx1.redhat.com [10.5.110.27]);
 Fri, 09 Nov 2018 13:59:51 +0000 (UTC)
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Since Qemu commit 9b4cf107b09d ("hyperv: only add SynIC in compatible
 configurations") hv_vpindex flag is mandatory if hv_synic is used:

$ qemu-system-x86_64 -nodefaults -cpu kvm64,hv_time,hv_synic,hv_stimer  \
     -device hyperv-testdev -device pc-testdev -device isa-debug-exit,  \
     iobase=0xf4,iosize=0x4 -vnc none -serial stdio -device pci-testdev \
     -machine accel=kvm -kernel x86/hyperv_stimer.flat

Results in:

Hyper-V SynIC (requested by 'hv-synic' cpu flag) requires Hyper-V VP_INDEX
 ('hv-vpindex')
qemu-system-x86_64: kvm_init_vcpu failed: Function not implemented

Fix the configuration.

Signed-off-by: Vitaly Kuznetsov <vkuznets@redhat.com>
---
 x86/unittests.cfg | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 3b21a85..c744a6d 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -620,19 +620,19 @@ arch = x86_64
 [hyperv_synic]
 file = hyperv_synic.flat
 smp = 2
-extra_params = -cpu kvm64,hv_synic -device hyperv-testdev
+extra_params = -cpu kvm64,hv_vpindex,hv_synic -device hyperv-testdev
 groups = hyperv
 
 [hyperv_connections]
 file = hyperv_connections.flat
 smp = 2
-extra_params = -cpu kvm64,hv_synic -device hyperv-testdev
+extra_params = -cpu kvm64,hv_vpindex,hv_synic -device hyperv-testdev
 groups = hyperv
 
 [hyperv_stimer]
 file = hyperv_stimer.flat
 smp = 2
-extra_params = -cpu kvm64,hv_time,hv_synic,hv_stimer -device hyperv-testdev
+extra_params = -cpu kvm64,hv_vpindex,hv_time,hv_synic,hv_stimer -device hyperv-testdev
 groups = hyperv
 
 [hyperv_clock]
